### PR TITLE
Extensible badges

### PR DESCRIPTION
**Pull request by Batmat**
This pull-request introduces two features, through extension points:
* one extension point to add/override badges for causes
* one extension point for deactivating badges for some causes

### CODE REVIEW

**Comment by Batmat** [comment]
@mpapo As this modification is a bit bigger than the general ones we do, I thought having a review would be cool.

@catalinpoitasu this PR supercedes the one you filed last september. I thought you might be interested in having a look. The extension point you want to have, IIRC, is BuildTriggerBadgeDeactivator.

Thanks for reviewing.

File: src/main/java/org/jenkinsci/plugins/buildtriggerbadge/IconFinder.java [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+import java.util.Map;
+
+import jenkins.model.Jenkins;
+
+import org.jenkinsci.plugins.buildtriggerbadge.provider.BuildTriggerBadgeProvider;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class responsible for finding the icon assocated with a build cause.
```
Comment (mpapo) [inline src/main/java/org/jenkinsci/plugins/buildtriggerbadge/IconFinder.java]: associated

**Comment by mpapo** [comment]
:+1: Good job :-)

**Comment by Batmat** [comment]
I'm gonna release a beta version just now. I'll have a try myself with a tentative extension, to see at least it does not crash the server ;-).

**Pull request merged**